### PR TITLE
Implements Detailed Plan View with Edit Mode Button

### PR DESCRIPTION
**Pull request by katrinan029**
# Description
As part of enhancing the user experience in the provisioning tool for LC2, we need to implement a detailed plan view that displays the plan details based on the original form inputs. Additionally, this detailed plan view should include an "Edit plan" button that enables them to make changes to the plan details.

# Solution
Add a detail view with an Edit Plan button. Note that the edit button has no functionality since that will be added in a follow up ticket.

This PR is part of an epic to implement an edit/view learner credit plan details in support tools. Below are the previous tickets associated:

|Implementation|PR|
| ------------- | ------------- |
|Add feature flag| https://2u-internal.atlassian.net/browse/ENT-7517| 
|Create List View of Learner Credit Plans with Muted Standalone Hyperlinks|https://github.com/openedx/frontend-app-support-tools/pull/349|

## Detail View with multiple policies
Screenshot 2023-08-23 at 5 24 05 PM
Screenshot 2023-08-23 at 5 24 15 PM
Screenshot 2023-08-23 at 5 24 25 PM

## Detail View with a single policy
Screenshot 2023-08-23 at 5 57 58 PM
Screenshot 2023-08-23 at 5 58 07 PM

[JIRA](https://2u-internal.atlassian.net/browse/ENT-7504)

### CODE REVIEW

File: src/Configuration/Provisioning/SubsidyDetailView/PolicyDetailView/CustomCatalogDetail.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+const { FORM: { CUSTOM_CATALOG } } = PROVISIONING_PAGE_TEXT;
+
+function getCustomCatalogTitle(catalogTitle) {
+  let index;
+  if (catalogTitle.includes(' - ')) {
+    index = catalogTitle.indexOf(' - ');
+    return catalogTitle.slice(index + 3);
+  }
+  return catalogTitle;
```
Comment (marlonkeating) [inline src/Configuration/Provisioning/SubsidyDetailView/PolicyDetailView/CustomCatalogDetail.jsx]: nit: Can reduce code duplication
```suggestion
  let separator = ' - ';
  if (catalogTitle.includes(separator)) {
    return catalogTitle.split(separator)[1];
  }
  return catalogTitle;
```

File: src/Configuration/Provisioning/SubsidyDetailView/PolicyDetailView/AssociatedCatalogDetail.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  return (
+    <div className="mb-1 mt-3">
+      <h3>{CATALOG.TITLE}</h3>
+      <div className="mb-1 ml-3 mt-3">
+        <h4>{CATALOG.SUB_TITLE}</h4>
+        <p className="small">
+          {catalogType}
+        </p>
+        {(catalogType === 'Custom') && <CustomCatalogDetail catalogTitle={associatedCatalog} />}
```
Comment (marlonkeating) [inline src/Configuration/Provisioning/SubsidyDetailView/PolicyDetailView/AssociatedCatalogDetail.jsx]:
```suggestion
        {(catalogType === CATALOG.OPTIONS.custom) && <CustomCatalogDetail catalogTitle={associatedCatalog} />}
```

File: src/Configuration/Provisioning/SubsidyDetailView/SubsidyDetailView.jsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            ...customer.data[0],
+          },
+          policies: policies.data.results.filter(policy => policy.subsidy_uuid === subsidyUuid),
+        }));
+
+        let catalogs;
+        if (catalogUuids !== undefined) {
+          catalogs = await Promise.all(catalogUuids.map(async (catalogUuid) => {
+            const catalog = await getCatalogs(catalogUuid);
+            return Object.assign({}, ...catalog.data.results);
```
Comment (marlonkeating) [inline src/Configuration/Provisioning/SubsidyDetailView/SubsidyDetailView.jsx:93]: Don't think we need Object.assign here?
```suggestion
            return {...catalog.data.results};
```

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: src/Configuration/Provisioning/SubsidyDetailView/PolicyDetailView/CustomCatalogDetail.jsx (hunk 1 above)
Comment (katrinan029) [reply src/Configuration/Provisioning/SubsidyDetailView/PolicyDetailView/CustomCatalogDetail.jsx]: good call!

File: src/Configuration/Provisioning/SubsidyDetailView/SubsidyDetailView.jsx (hunk 3 above)
Comment (katrinan029) [reply src/Configuration/Provisioning/SubsidyDetailView/SubsidyDetailView.jsx:93]: The reason I am using the `Object.assign` here is to remove the index. See differences below. 
Screenshot 2023-08-28 at 10 27 43 AM

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**